### PR TITLE
Split the remove measurements pass into its own source file.

### PR DESCRIPTION
**Pull request by schweitzpgi**
~Apply pass in parallel.~

Remove dead code.

### CODE REVIEW

File: runtime/cudaq/platform/default/rest/helpers/ionq/ionq.config [hunk 1]
```diff
@@ -23,7 +23,7 @@ CODEGEN_EMISSION=qir-base
 
 # Additional passes to run after lowering to QIR
 # This is required due to https://github.com/NVIDIA/cuda-quantum/issues/512
-POST_CODEGEN_PASSES="remove-measurements"
+POST_CODEGEN_PASSES="func.func(remove-measurements),llvm.llvmfunc(remove-measurements)"
```
Comment (bmhowe23) [inline runtime/cudaq/platform/default/rest/helpers/ionq/ionq.config:26]: What does `llvm.llvmfunc()` do here?

Comment (bmhowe23) [reply runtime/cudaq/platform/default/rest/helpers/ionq/ionq.config:26]: Does it run `func.func(...)` in parallel with `llvm.llvmfunc(...)`, or does it just run each one sequentially? (And each one can run on the corresponding ops in parallel?)

Comment (schweitzpgi) [reply runtime/cudaq/platform/default/rest/helpers/ionq/ionq.config:26]: IIRC, the pass manager can run them both in parallel.

Comment (schweitzpgi) [reply runtime/cudaq/platform/default/rest/helpers/ionq/ionq.config]: Alas, making this parallel was causing the async tests to fail with the ionq backend. I'll file an issue.

**Review by bmhowe23 — APPROVED** [review]

**Pull request merged**